### PR TITLE
Add semanticdb-javac to `TestInternal`

### PR DESCRIPTION
**Pull request by armanbilge**
So far, added a test to reproduce the problem. Now, let's see if I can fix it too 🙃 

Edit: done. Fixes https://github.com/scalameta/metals/issues/3903

### CODE REVIEW

File: sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala [hunk 1]
```diff
@@ -72,8 +72,8 @@ object MetalsPlugin extends AutoPlugin {
     },
     allDependencies ++= {
       if (javaSemanticdbEnabled.value)
-        List(
-          "com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion % Configurations.CompileInternal
+        List(Configurations.CompileInternal, Configurations.TestInternal).map(
+          "com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion % _
```
Comment (dos65) [inline sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala:76]: Tried it locally and it's need also to cover Integrations tests.
```suggestion
      val existing = ivyConfigurations.value.map(_.name).toSet
      val configurations = Seq(
        Configurations.CompileInternal,
        Configurations.TestInternal,
        Configurations.IntegrationTestInternal
      ).map(_.name).filter(existing.contains).mkString(",")
       List(
          "com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion % configurations
        )
```

**Review by dos65 — COMMENTED** [review]
Thanks for the fix!

File: sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala (hunk 1 above)
Comment (armanbilge) [reply sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala]: Thanks, good point!

**Review by dos65 — APPROVED** [review]

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**